tests/integration-tests/src/appinventory.rs

```diff
@@ -0,0 +1,204 @@
+        .expect("failed to get response from github api for release list");
+    let release_list: Vec<GithubRelease> =
+        serde_json::from_slice(&response_body).expect("malformed data returned from github api");
+    let mut latest: Option<String> = None;
+    for release in release_list.iter() {
+        if release.tag_name.contains("-rc") {
+            continue;
+        }
+        if let Some(current) = latest.as_ref() {
+            if release.tag_name > *current {
+                latest = Some(release.tag_name.clone());
+            }
+        } else {
+            latest = Some(release.tag_name.clone());
+        }
+    }
+    latest.expect(format!("could not find latest version from github for {repository}").as_str())
+}
+
+async fn create_bob(version: &str) -> TempDir {
+    let tmp_dir = TempDir::new().expect("failed to create temporary directory");
+    let git = which("git").expect("failed to find git");
+    let inside = tmp_dir.path().to_str().unwrap();
+    cmd!(
+        git,
+        "clone",
+        "-b",
+        version,
+        "https://github.com/bottlerocket-os/bottlerocket.git",
+        inside
+    )
+    .run()
+    .expect("failed to clone bob");
+    tmp_dir
+}
+
+// Installs twoliter from the latest non-rc release on github
+async fn install_twoliter_from_release() -> (TempDir, PathBuf) {
+    let tmp_dir = TempDir::new().expect("failed to create temporary directory");
+    let twoliter_bin = tmp_dir.path().join("twoliter");
+    let latest = find_latest_version("twoliter").await;
+    let arch = env::consts::ARCH;
+    let response = reqwest::get(format!("https://github.com/bottlerocket-os/twoliter/releases/download/{latest}/twoliter-{arch}-unknown-linux-musl.tar.xz")).await.expect("failed to download twoliter release archive");
+    let mut archive_bytes = BufReader::new(Cursor::new(
+        response
+            .bytes()
+            .await
+            .expect("failed to fetch twoliter release archive"),
+    ));
+    let mut decompressed_bytes: Vec<u8> = Vec::new();
+    lzma_rs::xz_decompress(&mut archive_bytes, &mut decompressed_bytes)
+        .expect("failed to decompress archive");
+    let mut archive = Archive::new(Cursor::new(decompressed_bytes));
+    let mut twoliter = File::create(&twoliter_bin).expect("failed to create twoliter binary");
+    // Now iterate through the archive till we find the twoliter binary
+    let mut entries = archive.entries().expect("failed to get archive iterator");
+    while let Some(Ok(mut entry)) = entries.next() {
+        if entry.path().unwrap().ends_with("twoliter") {
+            std::io::copy(&mut entry, &mut twoliter).expect("failed to write twoliter executable");
+        }
+    }
+    // Make the file executable
+    cmd!("chmod", "+x", &twoliter_bin)
+        .run()
+        .expect("failed to make twoliter executable");
+    (tmp_dir, twoliter_bin)
+}
+
+#[derive(Deserialize, Eq, Debug)]
+#[serde(rename_all = "PascalCase")]
+struct ContentView {
```

Comment:
Not a hard blocker, but I'd prefer to hoist this as `pub` into something that we could re-use to perform inventory generation in Rust.